sdk/metric/meter.go

```diff
@@ -429,17 +437,15 @@ func (p float64ObservProvider) registerCallbacks(inst *observable[float64], cBac
 	}
 }
 
-func (p float64ObservProvider) callback(i *observable[float64], f instrument.Float64Callback) func(context.Context) error {
-	inst := callbackObserver[float64]{i}
+func (p float64ObservProvider) callback(i float64Observable, f instrument.Float64Callback) func(context.Context) error {
+	inst := float64Observer{i}
 	return func(ctx context.Context) error { return f(ctx, inst) }
 }
 
-// callbackObserver is an observer that records values for a wrapped
-// observable.
-type callbackObserver[N int64 | float64] struct {
-	*observable[N]
+type float64Observer struct {
+	float64Observable
 }
 
-func (o callbackObserver[N]) Observe(_ context.Context, val N, attrs ...attribute.KeyValue) {
+func (o float64Observer) Observe(val float64, attrs ...attribute.KeyValue) {
 	o.observe(val, attrs)
 }
```

Comment:
Is there a need for this type? couldn't this be a function on float64Observable or *observable[N] and then we can use them directly?